### PR TITLE
Implementation of backend in quizzess component

### PR DESCRIPTION
**Pull request by nitesh2920**

### CODE REVIEW

File: client/src/App.js
Comment (RohitRaj766) [inline client/src/App.js:1]: quiz page should be a private route only logged in user can visit this route.

Comment (RohitRaj766) [reply client/src/App.js:1]: do not add unnecessary comment lines in app file.

File: client/src/actions/index.js
Comment (RohitRaj766) [inline client/src/actions/index.js:1]: remove all the unnecessary changes you have done. Please stick to the task given do not alter the previously written codes.

Comment (RohitRaj766) [reply client/src/actions/index.js:1]: Use correct naming conventions.
If you are fetching data for quizlist keep it sound similar not like this FETCH_DATA_REQUEST ( by this name we cannot identify that what data it is getting.)
similarly if you are fetching questions of quizzes then then it should also sound similar.

File: client/src/reducers/userauthReducer.js
Comment (RohitRaj766) [inline client/src/reducers/userauthReducer.js:1]: remove all the unnecessary changes you have done. Please stick to the task given do not alter the previously written codes.

Comment (RohitRaj766) [inline client/src/reducers/userauthReducer.js:1]: Give Correct Names.
(fetchedData) = what do it means ? 
all data is fetched only so give correct name.

File: client/src/sagas/fetchquizdataSaga.js
Comment (RohitRaj766) [inline client/src/sagas/fetchquizdataSaga.js:1]: File name will be quiz list not quiz data.

File: client/src/sagas/index.js
Comment (RohitRaj766) [inline client/src/sagas/index.js:1]: quiz data will be quiz list (quiz questions are also the part of quiz only but they are not fetched in quizdata {understand these things.})
Remove watch. Please see the previously written code and make yours accordingly.

File: client/src/user/userScreens/dashboard/mocktest/Mocktest.jsx
Comment (RohitRaj766) [inline client/src/user/userScreens/dashboard/mocktest/Mocktest.jsx:1]: Why you have made changes in the mocktest file ? this task is not associated with mocktest.
This task is seperate entity. please remove the unnecessary changes you have made to this page.

File: client/src/user/userScreens/dashboard/quiz/Quizzes.jsx
Comment (RohitRaj766) [inline client/src/user/userScreens/dashboard/quiz/Quizzes.jsx:1]: Too many commented lines are there please remove it.

File: client/src/App.js
Comment (nitesh2920) [reply client/src/App.js:1]: > quiz page should be a private route only logged in user can visit this route.

only logged in users can play a quiz it is already implemented in that way only see the fetchquiz question saga

Comment (RohitRaj766) [reply client/src/App.js:1]: Route is not protected it should be under protected routes.

Comment (RohitRaj766) [reply client/src/App.js:1]: > > quiz page should be a private route only logged in user can visit this route.
> 
> only logged in users can play a quiz it is already implemented in that way only see the fetchquiz question saga

Route is not protected it should be under protected routes.

Comment (nitesh2920) [reply client/src/App.js:1]: Only the login user can access that route if any other try to access they are redirect to login.

Comment (nitesh2920) [reply client/src/App.js:1]: changes have been made.

**Review by RohitRaj766 — APPROVED** [review]
